origin-dapp/src/components/customize.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { injectIntl } from 'react-intl'
+import { withRouter } from 'react-router'
+
+import { fetchConfig } from 'actions/Config'
+import store from 'store'
+
+class Customize extends React.Component {
+
+  componentDidMount() {
+    this.getConfigUrl = this.getConfigUrl.bind(this)
+    this.setConfig = this.setConfig.bind(this)
+
+    const configUrl = this.getConfigUrl()
+
+    this.props.fetchConfig(configUrl)
+      .then(this.setConfig)
+      .catch((error) => {
+        console.log('Could not load custom configuration: ' + error)
+      })
+  }
+
+  getConfigUrl () {
+    // Config override specified as URL parameter
+    const configUrl = this.getConfigOverrideUrl()
+    if (configUrl) {
+      return configUrl
+    } else if (this.isWhiteLabelHostname()) {
+      // Retrieve the config from config.hostname
+      const ipfsGateway = `${process.env.IPFS_GATEWAY_PROTOCOL}://${process.env.IPFS_DOMAIN}:${process.env.IPFS_GATEWAY_PORT}`
+      return `${ipfsGateway}/ipns/config.${window.location.hostname}`
+    }
+  }
+
+  getConfigOverrideUrl () {
+    const configUrlMatch = window.location.search.match(/config=([^#]*)/)
```

Comment:
This was (on my part) a pretty hacky way to get the query string var and it's not very smart. I.e. if `\config=XXX` appears _anywhere_ in the URL it will get parsed. Is this open to abuse/security risk? The "right" way I guess would be to use something like:
https://www.npmjs.com/package/url-parse

Reply:
It'd be cleaner though I think, this is a bit ugly. But it should be enough for now.

I was really surprised React router doesn't do its own URL parameter parsing!